grpcurl.go

```diff
@@ -161,6 +163,37 @@ func MetadataFromHeaders(headers []string) metadata.MD {
 	return md
 }
 
+/* Expands environmental variables contained in the header string
+ * If no corresponding environmental variable is found, the header
+ * string is not changed. Hence, if the regex matches accidentally
+ * no changes are made.
+ */
+func ExpandHeaders(headers []string) []string {
+	expandedHeaders := make([]string, len(headers))
+	for idx, header := range headers {
+		if header != "" {
+			regex := regexp.MustCompile(`\${\w+}`)
+			results := regex.FindAllString(header, -1)
+			if results != nil {
+				expandedHeader := header
+				for _, result := range results {
+					envVarValue := os.Getenv(result[2 : len(result)-1])
+					replacementValue := envVarValue
+					// If no corresponding env var is found, leave the header as is.
+					if len(envVarValue) == 0 {
```

Comment:
I think this will be confusing. Based on how other tools generally work with env vars, I would expect either an error that my input is bad or missing env vars to interpolate the empty string.

I'm a bit partial to the former since it is more likely catch bugs.

Although this does bring up an interesting point: if you want to use a literal `${` in the value, this scheme is too simple and would result in an error. I think we need a way to escape the `$`. For example, `$$` or `\$` meaning literal dollar sign, not the start of an env var reference.

Reply:
Escaping can probably be left as a TODO since, by default, grpcurl does not try to do any variable expansion. So you can either do variable expansion or use literal dollar signs, which is likely sufficient for now.

Reply:
Agreed, on the confusing behavior. Fixed.